### PR TITLE
5.0.3  Timer changes

### PR DESCRIPTION
**Pull request by emveepee**
Start and end times not being set correctly on any time type 1 and 2.    Set the original time on timeslot recording.

### CODE REVIEW

**Comment by emveepee** [comment]
@sub3 when you have a chance after approval from @phunkyfish could you tag and merge this.  In Matrix this is critical because of a crash,  Somewhere Matrix is doing a local time to UTC conversion  with a  local time of 0  which is failing in UTC+ timezones.  This logic improvement avoids the root cause of the crash.  I will also port this to Leia as it is a more accurate representation but Leia is not crashing.

Note that after the C++ port I will make the rules names in Kodi look more like the v5 web.

**Review by phunkyfish — APPROVED** [review]

**Pull request merged**
